src/napari_threedee/manipulators/point_manipulator.py

```diff
@@ -1,201 +1,130 @@
+            # toggling following line on/off breaks magnitude of translator vector
+            # but is needed to update position of manipulator when selection is changed...
+            self.origin = self.active_point_position
+            print('origin updated')
         else:
             self.visible = False
-            if self._layer.mode == 'select':
+            if self.layer.mode == 'select':
                 remove_mouse_callback_safe(
-                    self._layer.mouse_drag_callbacks,
+                    self.layer.mouse_drag_callbacks,
                     self.napari_selection_callback_passthrough
                 )
                 add_mouse_callback_safe(
-                    self._layer.mouse_drag_callbacks, napari_selection_callback
+                    self.layer.mouse_drag_callbacks, napari_selection_callback
                 )
 
-    def _pre_drag(
-            self,
-            click_point: np.ndarray,
-            selected_translator: Optional[int],
-            selected_rotator: Optional[int]
-    ):
-        pass
-
-    def _while_dragging_translator(self, selected_translator: int, translation_vector: np.ndarray):
-        self._layer._move([self.active_point_index], self.translation)
-        self._drag_start = None
+    def _pre_drag(self):
+        self.layer.events.highlight.disconnect(self._on_selection_change)
 
-    def _while_dragging_rotator(self, selected_rotator: int, rotation_matrix: np.ndarray):
-        # todo: store rotmat somewhere
-        pass
+    def _while_dragging_translator(self):
+        selected_point_index = list(self.layer.selected_data)[0]
+        self.layer.data[selected_point_index] = self.origin
+        # refresh rendering manually after modifying array data inplace
+        self.layer.refresh()
 
     def _post_drag(self):
-        pass
+        self.layer.events.highlight.connect(self._on_selection_change)
 
-    def napari_selection_callback_passthrough(self, layer, event):
-        # get click position and direction in data coordinates
-        click_position_world = event.position
-
-        click_position_data_3d = np.asarray(
-            self._layer._world_to_displayed_data(
-                click_position_world,
-                event.dims_displayed
-            )
-        )
-        click_dir_data_3d = np.asarray(
-            self._layer._world_to_displayed_data_ray(
-                event.view_direction,
-                event.dims_displayed
-            )
-        )
 
-        # identify clicked rotator/translator
-        selected_translator, selected_rotator = self._check_if_manipulator_clicked(
-            plane_point=click_position_data_3d,
-            plane_normal=click_dir_data_3d,
-        )
-        manipulator_clicked = (selected_translator is not None) or (selected_rotator is not None)
+    # def _while_dragging_rotator(self, selected_rotator: int, rotation_matrix: np.ndarray):
+    #     # todo: store rotmat somewhere
+    #     pass
```

Comment:
can you please implement this?

Reply:
tried and ran into a few problems, making issue for followup